docs/user-documentation/moderne-platform/how-to-guides/writing-and-installing-recipes.md

```diff
@@ -0,0 +1,38 @@
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Writing and Installing Recipes
+
+[OpenRewrite](https://docs.openrewrite.org/) is a framework for writing and running code transformations. A new code transformation, called [recipes](https://docs.openrewrite.org/concepts-and-explanations/recipes) can be written
+and uploaded to your instance of the Moderne Platform. This is done in 3 steps:
+
+1. Writing custom recipes
+2. Publish artifact containing the custom recipes
+3. Importing Recipes
+
+## Writing Recipes
+Before writing a new recipe, check the [recipe catalogue](https://docs.openrewrite.org/recipes) to see if an existing recipe is available. For transformations that aren't covered by the existing catalogue, new recipes can be written in:
+ - [Declarative YAML](https://docs.openrewrite.org/concepts-and-explanations/recipes#declarative-recipes)
+ - [Refaster](https://docs.openrewrite.org/authoring-recipes/refaster-recipes)
+ - [Imperative Java](https://docs.openrewrite.org/concepts-and-explanations/recipes#imperative-recipes).
+
+As a best practice, if your recipe can be built out of other recipes, you should make it declarative. You can make some powerful migration recipes by
+combining many tiny recipes. Depending on the code transformation a declarative YAML recipe might not be possible, and using an imperative Java or a 
+Refaster recipe would be more appropriate. For details on how to write recipes follow these [instructions](https://docs.openrewrite.org/authoring-recipes/types-of-recipes). Try our [Recipe authoring workshop](../../workshops/recipe-authoring.md) 
+for a step-by-step example of recipe authoring.
+
+## Creating a Recipe Jar
+
+After the new recipes have been written, they need to be published to an Artifactory which has been set up as a [recipe source](../../../administrator-documentation/moderne-platform/how-to-guides/agent-configuration/configure-an-agent-with-maven-repository-access.md#publishing-recipe-artifacts).
+
+You must set up your repository to publish the new recipes to your Artifactory or Maven Central. The [rewrite-recipe-starter](https://github.com/moderneinc/rewrite-recipe-starter/blob/main/build.gradle.kts)
+project can serve as an example of how to publish an artifact. This repository is configured to publish to Moderne's open artifact repository (via the publishing task at the bottom of the `build.gradle.kts` file).1
```

Comment:
To any Maven repository. Given that this is in the Moderne docs, we can list out all the Maven repos that the Moderne SaaS has access to: https://github.com/moderneinc/moderne-worker-model/blob/8c717ea92d6f871d27e90352817d89bb62a9f5c0/src/main/java/io/moderne/worker/AgentRepositorySettings.java#L89-L115

I also thought we had a no-auth maven repo in our Artifactory (`moderne-recipe`) where folks could publish arbitrary recipe jars? But I see that was removed in https://github.com/moderneinc/moderne-worker-model/pull/342 .